### PR TITLE
Android SDK adding README.md

### PR DESCRIPTION
**Pull request by daniftodi**
extole_android_sdk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+extole.event("purchase", mapOf(Pair("partner_user_id", "123"), Pair("cart_value", "120.3")))
+```
+
+### Get a text configured in your Extole Creative
+
+```
+val emailShareMessage = extole.getText("sharing.email.message")
+```
+
+### Sharing via Email
```
Comment (mcyster) [inline README.md]: Sharing via Email through Extole

File: README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            .build()
+```
+
+### Sending events
+
+```
+extole.event("purchase", mapOf(Pair("partner_user_id", "123"), Pair("cart_value", "120.3")))
+```
+
+### Get a text configured in your Extole Creative
```
Comment (mcyster) [inline README.md]: Get marketer configured campaign attributes

File: README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+## Demo application
+Demo application is very simple, it shows basic operations like:
+- sending an event
+- email share
+- getting a resource (text or url) from your configured Extole Creative
+- native sharing
+
+Screenshot:
+
+<img src="https://user-images.githubusercontent.com/304224/130804856-ee7b5404-4a8d-4975-ad17-85cc0bf6e253.png" width="550">
```
Comment (mcyster) [inline README.md]: I would prefer this was smaller, say width 150 

When the app is bigger we can show a number of images of the app, linking to the specific code in the repo that does the functionality

File: README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            .addAppData("version", "1.0")
+            .withSandbox("prod-test")
+            .withDebugEnabled(true)
+            .build()
+```
+
+### Sending events
+
+```
+extole.event("purchase", mapOf(Pair("partner_user_id", "123"), Pair("cart_value", "120.3")))
```
Comment (mcyster) [inline README.md]: 120.30

Future: This looks clumsy (I wonder if we should have an event builder)

File: README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+```
+val emailShareMessage = extole.getText("sharing.email.message")
+```
+
+### Sharing via Email
+
+```
+val eventId = extole.emailShare("shareToEmailAddress", emailShareMessage)
+```
+
```
Comment (mcyster) [inline README.md:41]: Native Shareing

File: README.md (hunk 2 above)
Comment (daniftodi) [reply README.md]: changed

File: README.md (hunk 1 above)
Comment (daniftodi) [reply README.md]: changed

File: README.md (hunk 3 above)
Comment (daniftodi) [reply README.md]: changed

File: README.md (hunk 4 above)
Comment (daniftodi) [reply README.md]: changed.
Builder - probably yes, let's discuss

File: README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+## Demo application
+Demo application is very simple, it shows basic operations like:
```
Comment (mcyster) [inline README.md:50]: The demo application illustrates how to leverage the Extole Android SDK to do things like:

File: README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+### Sharing via Email through Extole
+
+```
+val eventId = extole.emailShare("shareToEmailAddress", emailShareMessage)
+```
+
+### Native Sharing
+
+```
+extole.nativeShare("title", "shareMessage")
```
Comment (mcyster) [inline README.md:45]: This is not right, 
- emailShare - does share
- native share does not do a share

extole.shareIntentBuilder
.withView()
.withDefaultMessage()
.withLink()
...
.withXXX
.setup()

it should 
- setup the share intent
- generate a partner_share_id
- fire a share invent specifying native share


extole.setup
// tbd also have that actually 
extole.fireNativeShareEvent()


// ALSO need to add (future)
extole.fireCTAViewed / fireCTATapped, ...

File: README.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+            .addAppData("version", "1.0")
+            .withSandbox("prod-test")
+            .withDebugEnabled(true)
+            .build()
+```
+
+### Sending events
+
+```
+extole.event("purchase", mapOf(Pair("partner_user_id", "123"), Pair("cart_value", "120.30")))
```
Comment (mcyster) [inline README.md:27]: Feel we should get to here
extole.eventBuilder("name")
.withParameter(name, value)
.send()

**Review by mcyster — COMMENTED** [review]
I merged this, we can iterate.

**Pull request merged**